### PR TITLE
Add SnapConfig.unset

### PR DESCRIPTION
**Pull request by gnuoy**
Add SnapConfig.unset
    
Add SnapConfig.unset to allow for config keys to be unset via the
SnapConfig object.

### CODE REVIEW

File: integration_tests.py [hunk 1]
```diff
@@ -33,3 +34,38 @@ def test_service(snap: Snap, service: str):
...
     assert services[service].active
+
+
+def test_config_set_unset(snap: Snap):
+    # Unset the test keys before running the test proper just in case
+    # it has been set by another test.
+    try:
+        snap.config.unset(['foo', 'goo'])
+    except UnknownConfigKey:
+        pass
```
Comment (sparkiegeek) [inline integration_tests.py]: use `contextlib.suppress` for ease of reading?

```suggestion
    with suppress(UnknownConfigKey):
        snap.config.unset(['foo', 'goo'])
```

**Review by sparkiegeek — CHANGES_REQUESTED** [review]
Don't think you want the .snap added to the repo?

File: integration_tests.py [hunk 2]
```diff
@@ -33,3 +35,36 @@ def test_service(snap: Snap, service: str):
     assert service in services
     assert services[service].enabled
     assert services[service].active
+
+
+def test_config_set_unset(snap: Snap):
+    # Unset the test keys before running the test proper just in case
+    # it has been set by another test.
+    with contextlib.suppress(UnknownConfigKey):
```
Comment (albertodonato) [inline integration_tests.py]: this shouldn't be needed, since unset won't fail if the config doesn't exist

File: integration_tests.py [hunk 3]
```diff
@@ -33,3 +35,36 @@ def test_service(snap: Snap, service: str):
...
+        snap.config.unset(['foo', 'goo'])
+    # Start Test:
```
Comment (albertodonato) [inline integration_tests.py]: this comment could be dropped

**Review by albertodonato — CHANGES_REQUESTED** [review]
Thanks for the changes! 

Just a couple of nits inline.

Also please run `tox -e format` to ensure formatting, and a unit test for `unset` in `snaphelpers/tests/test_conf.py` (similar to the one for `set`) would also be needed.

**Review by albertodonato — APPROVED** [review]
Thanks for fixing CI issues!

I'll take a look at running integration tests as root separately, so that's possible to add tests like the one you had.

**Pull request merged**